chart/openfaas/templates/secret.yaml

```diff
@@ -0,0 +1,19 @@
+{{- if .Values.basicAuthAutogen }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: basic-auth
+  namespace: {{ .Release.Namespace | quote }}
+  labels:
+    app: {{ template "openfaas.name" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+    component: gateway
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+  annotations:
+    "helm.sh/hook": "pre-install"
+data:
+  basic-auth-user: {{ "admin" | b64enc | quote }}
+  # kubectl -n openfaas get secret basic-auth -o jsonpath="{.data.basic-auth-password}" | base64 --decode
+  basic-auth-password: {{ randAlphaNum 12 | b64enc | quote }}
```

Comment:
Pretty :+1: 

What happens during upgrades though?

Reply:
It's a pre-install hook, Helm will not run it on upgrades, the password will stay the same.

Reply:
Is it ignored for `helm template`?

Reply:
`generateBasicAuth` is false by default so it will not end up in the template output